network.go

```diff
@@ -777,44 +786,76 @@ func (n *network) getController() *controller {
 	return n.ctrlr
 }
 
-func (n *network) ipamAllocate() ([]func(), error) {
-	var (
-		cnl []func()
-		err error
-	)
-
+func (n *network) ipamAllocate() error {
 	// For now also exclude bridge from using new ipam
 	if n.Type() == "host" || n.Type() == "null" || n.Type() == "bridge" {
-		return cnl, nil
+		return nil
 	}
 
 	ipam, err := n.getController().getIpamDriver(n.ipamType)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	if n.addrSpace == "" {
 		if n.addrSpace, err = n.deriveAddressSpace(); err != nil {
-			return nil, err
+			return err
+		}
+	}
+
+	err = n.ipamAllocateVersion(4, ipam)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			n.ipamReleaseVersion(4, ipam)
 		}
+	}()
+
+	return n.ipamAllocateVersion(6, ipam)
+}
+
+func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
+	var (
+		cfgList  *[]*IpamConf
+		infoList *[]*IpamInfo
+		err      error
+	)
+
+	switch ipVer {
+	case 4:
+		cfgList = &n.ipamV4Config
```

Comment:
That's why I said it's merely a suggestion. You can ignore if you don't want to do that and you personally like operating with pointers to slices. 

But code review is not just about checking if functionality is working. If that is case we should not make any comments about how the code should be organized or how a function should be named or how the logic should be simplified for future maintenance even if it was already achieving the expected functionality